coalib/results/Diff.py

```diff
@@ -278,18 +276,37 @@ def modified(self):
         return result
 
     @property
+    def modified(self):
+        """
+        Calculates the modified file, after applying the Diff to the original.
+        Also adds linebreaks at the end of each line (except eof)
+        """
+        eof_linebreak = False
+
+        if self._file and self._file[-1].endswith('\n'):
+            eof_linebreak = True
+
+        file = [line.rstrip('\n') for line in self._modified()]
+
+        if eof_linebreak:
+            return self._add_linebreaks(file)
+        return self._generate_linebreaks(file)
+
+    @property
     def unified_diff(self):
         """
         Generates a unified diff corresponding to this patch.
 
         Note that the unified diff is not deterministic and thus not suitable
         for equality comparison.
         """
-        return ''.join(difflib.unified_diff(
-                self.original,
-                self.modified,
+        list_unified_diff = list(difflib.unified_diff(
+                self._file,
+                self._modified(),
```

Comment:
In a case with "eof-state" without `\n`, this would result in the last line of the original/changed file being on the same line in the diff, something like:
`original = ['1','2','3']`
`target = ['1','2','4']`
`diff = ['+++\n', '---\n', '@@ xxxxx @@\n', '1\n', '2\n', '-3+4']`

I found this weird, because each change should be on a new line.
So I have to run either `_generate_linebreaks` or `_add_linebreaks` anyway to avoid this problem.
(However, the latter is unnecessary, because it doesn't matter if the last line of the diff has a '\n' or not - the representation seems equal).

By using the un-linebreaked files for unified_diff creation and add linebreaks later, I avoid unnecessary calculations.